internal/pkg/vault/secrets.go

```diff
@@ -440,9 +439,7 @@ func (c *Client) store(subPath string, secrets map[string]string) error {
 		return err
 	}
 	defer func() {
-		if resp.Body != nil {
-			_ = resp.Body.Close()
-		}
+		_ = resp.Body.Close()
```

Comment:
Was was the resp.Body nil check removed here?

Reply:
According to this [official http client documentation](https://golang.org/pkg/net/http/#Client.Do), If the returned error is nil, the Response will contain a non-nil Body

Reply:
indeed.

Reply:
> If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close. If the Body is not both read to EOF and closed, the Client's underlying RoundTripper (typically Transport) may not be able to re-use a persistent TCP connection to the server for a subsequent "keep-alive" request.


> On error, any Response can be ignored. A non-nil Response with a non-nil error only occurs when CheckRedirect fails, and even then the returned Response.Body is already closed.


According to the above, on success, we need to close the Body explicitly.  On error, the body is already closed.

As long as the double close doesn't result in a panic(), I am ok with this.